src/sign-typed-data.test.ts

```diff
@@ -1,17 +1,232 @@
 import * as ethUtil from 'ethereumjs-util';
+import Ajv from 'ajv';
 import {
   recoverTypedSignature,
   signTypedData,
   TypedDataUtils,
   typedSignatureHash,
   SignTypedDataVersion,
+  TYPED_MESSAGE_SCHEMA,
 } from './sign-typed-data';
 
 const privateKey = Buffer.from(
   '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0',
   'hex',
 );
 
+/**
+ * Get a list of all Solidity types supported by EIP-712.
+ *
+ * @returns A list of all supported Solidity types.
+ */
+function getEip712SolidityTypes() {
+  const types = ['bool', 'address', 'string', 'bytes'];
+  const ints = Array.from(new Array(32)).map(
+    (_, index) => `int${(index + 1) * 8}`,
+  );
+  const uints = Array.from(new Array(32)).map(
+    (_, index) => `uint${(index + 1) * 8}`,
+  );
+  const bytes = Array.from(new Array(32)).map(
+    (_, index) => `bytes${index + 1}`,
+  );
+
+  return [...types, ...ints, ...uints, ...bytes];
+}
+
+const eip712SolidityTypes = getEip712SolidityTypes();
+
+describe('TYPED_MESSAGE_SCHEMA', () => {
+  it('should match valid typed message', () => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(TYPED_MESSAGE_SCHEMA);
```

Comment:
[nit] What do you think about extracting this `validate` function since it's the same for every test?

Reply:
Good idea! Done in b5f0d877f72fb36fd41d64eec7a8cb248aba33f5